roles/wordpress-setup/templates/wordpress-site.conf.j2

```diff
@@ -7,7 +7,7 @@ server {
   listen 80;
   {% endif %}
 
-  server_name  {{ item.value.site_hosts | join(' ') }};
+  server_name  {% for host in item.value.site_hosts %} {{ host }} {% if item.value.multisite.subdomains %} *.{{ host }} {% endif %} {% endfor %};
```

Comment:
Without having followed the bigger picture of the multisite discussion (I might be missing something), I think what you have is the best way to handle it. 

One (untested) alternative that comes to mind is just to tell users that if they're using multisite, they should add their the subdomain wildcard version to the list of `site_hosts`:
```yaml
site_hosts:
  - site1.com
  - *.site1.com
  - site2.com
  - *.site2.com
```
Then you could just keep the original 
`server_name  {{ item.value.site_hosts | join(' ') }};`

The other idea is pretty bad and I haven't tested: 
`server_name  {{ item.value.site_hosts | join(' ') }} {% if item.value.multisite.subdomains %} *.{{ item.value.site_hosts | join(' *.') }} {% endif %};`